src/Analysis/Engine/Impl/Indexing/SymbolIndexWalker.cs

```diff
@@ -0,0 +1,263 @@
+            WalkAndDeclareAll(node.Tests);
+            WalkAndDeclare(node.ElseStatement);
+
+            return false;
+        }
+
+        public override bool Walk(TryStatement node) {
+            WalkAndDeclare(node.Body);
+            WalkAndDeclareAll(node.Handlers);
+            WalkAndDeclare(node.Else);
+            WalkAndDeclare(node.Finally);
+
+            return false;
+        }
+
+        private void AddVarSymbol(NameExpression node) {
+            if (node == null) {
+                return;
+            }
+
+            var kind = SymbolKind.Variable;
+
+            switch (node.Name) {
+                case "*":
+                case "_":
+                    return;
+                case var s when (_stack.Parent == null || _stack.Parent == SymbolKind.Class) && ConstantLike.IsMatch(s):
+                    kind = SymbolKind.Constant;
+                    break;
+            }
+
+            var span = node.GetSpan(_ast);
+
+            _stack.AddSymbol(new HierarchicalSymbol(node.Name, kind, span));
+        }
+
+        private (SymbolKind, string)? DecoratorExpressionToKind(Expression exp) {
+            switch (exp) {
+                case NameExpression ne when NameIsProperty(ne.Name):
+                case MemberExpression me when NameIsProperty(me.Name):
+                    return (SymbolKind.Property, FunctionKind.Property);
+
+                case NameExpression ne when NameIsStaticMethod(ne.Name):
+                case MemberExpression me when NameIsStaticMethod(me.Name):
+                    return (SymbolKind.Method, FunctionKind.StaticMethod);
+
+                case NameExpression ne when NameIsClassMethod(ne.Name):
+                case MemberExpression me when NameIsClassMethod(me.Name):
+                    return (SymbolKind.Method, FunctionKind.ClassMethod);
+            }
+
+            return null;
+        }
+
+        private bool NameIsProperty(string name) =>
+            name == "property"
+            || name == "abstractproperty"
+            || name == "classproperty"
+            || name == "abstractclassproperty";
+
+        private bool NameIsStaticMethod(string name) =>
+            name == "staticmethod"
+            || name == "abstractstaticmethod";
+
+        private bool NameIsClassMethod(string name) =>
+            name == "classmethod"
+            || name == "abstractclassmethod";
+
+        private void WalkAndDeclare(Node node) {
+            if (node == null) {
+                return;
+            }
+
+            _stack.EnterDeclared();
+            node.Walk(this);
+            _stack.ExitDeclaredAndMerge();
+        }
+
+        private void WalkAndDeclareAll(IEnumerable<Node> nodes) {
+            foreach (var node in nodes.MaybeEnumerate()) {
+                WalkAndDeclare(node);
+            }
+        }
+
+        private class SymbolStack {
+            private readonly Stack<(SymbolKind?, List<HierarchicalSymbol>)> _symbols;
```

Comment:
`Stack<(SymbolKind? kind, List<HierarchicalSymbol> symbols)> _symbols`

Reply:
Done, and the other case where this would help.